pkg/helm/controller/reconcile.go

```diff
@@ -80,14 +80,29 @@ func (r HelmOperatorReconciler) Reconcile(request reconcile.Request) (reconcile.
 		log.V(1).Info("Adding finalizer", "finalizer", finalizer)
 		finalizers := append(pendingFinalizers, finalizer)
 		o.SetFinalizers(finalizers)
-		err := r.Client.Update(context.TODO(), o)
+		if len(status.Conditions) == 0 {
+			status.SetCondition(types.HelmAppCondition{
+				Type:   types.ConditionInitializing,
+				Status: types.StatusTrue,
+			})
+		}
+		err := r.updateResourceStatus(o, status)
 		return reconcile.Result{}, err
 	}
+	status.RemoveCondition(types.ConditionInitializing)
```

Comment:
Should you keep the initializing condition? (the same way that pod keeps container creating condition) to show that the initialization was completed successfully?